helix-core/src/main/java/org/apache/helix/tools/ClusterSetup.java

```diff
@@ -176,6 +179,30 @@ public void close() {
     }
   }
 
+  public void addCluster(String clusterName, boolean overwritePrevious, boolean isCloudEnabled)
+      throws HelixException {
+    if (isCloudEnabled) {
+      String cloudID = System.getProperty(SystemPropertyKeys.CLOUD_ID);
```

Comment:
Let's discuss where the system property should be used. If you call the property key here, it means that the default value for Azure is put in the cfg2 of Helix-rest, which is not quite proper. These cloud features are not part of Helix Rest configuration. Ideally it should be in participant side, and put in our client's code cfg2 (e.g. Espresso storage node). 
In cluster creation, we just populate whatever we have into Zookeeper. The remaining is participant's work.

Reply:
Fixed based on our discussion.